packages/react/src/components/Selector/index.js

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import UiText from '../UiText';
+
+const Selector = React.forwardRef(
+  (
+    {
+      className,
+      disabled: isDisabled,
+      id,
+      onChange,
+      options,
+      defaultIndex,
+      ...props
+    },
+    ref,
+  ) => {
+    const [activeIndex, setActiveIndex] = useState(defaultIndex);
+
+    const handleOptionClick = (index, option) => {
+      if (isDisabled || option.disabled) return;
+      if (index === activeIndex) {
+        setActiveIndex(null);
+        onChange(null);
+        return;
+      }
+
+      setActiveIndex(index);
+      onChange(option.value);
+    };
+
+    return (
+      <div className='f-FormEl-wrapper'>
+        <UiText
+          as='div'
+          variant={UiText.VARIANTS.content}
+          className={cx('f-Selector', className, {
+            'is-disabled': isDisabled,
+          })}
+          disabled={isDisabled}
+          id={id}
+          name={id}
+          onChange={onChange}
+          ref={ref}
+          {...props}
+        >
+          {options.map((option, index) => (
+            <div
+              key={option.value}
+              role='button'
```

Comment:
Why not using a `<Button />` here?
But using one, you could also inherit from all its styles (active, disabled, etc)

Reply:
You're right, I can try that instead

Reply:
So I tried using `<Button />`. I think it doesn't help much because I still need to override the styles of the states and add a couple of missing ones. I'll leave it this way